nifi-registry/nifi-registry-core/nifi-registry-web-api/src/main/java/org/apache/nifi/registry/web/api/BucketFlowResource.java

```diff
@@ -295,6 +295,48 @@ public Response createFlowVersion(
         return Response.status(Response.Status.OK).entity(createdSnapshot).build();
     }
 
+    @POST
+    @Path("{flowId}/versions/preserveSourceProperties")
```

Comment:
I think David's suggestion was that we should add an optional query param to the existing create method, so we wouldn't need this whole new method and instead a client could specify `?preserveSourceProperties=true` 

@exceptionfactory is that what you were thinking?

Reply:
Yes, that's what I had in mind, not a separate path.